### PR TITLE
docs: Access Monitoring IaC selfhosted setup

### PR DESCRIPTION
**Pull request by smallinsky**
### What
Extend Access Monitoring docs for self-hosted setup.

### CODE REVIEW

File: docs/pages/access-controls/access-monitoring.mdx [hunk 1]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
+| `eu-central-1`    | AWS region                                               |
+| `1234567890`      | AWS account ID                                           |
+| `audit-long-term` | S3 bucket used for long-term storage                     |
+| `audit-transient` | S3 bucket used for transient storage                     |
+| `kms_id`          | KMS key ID used for server-side encryption of S3         |
+| `audit_db`        | Glue database used for audit logs                        |
+| `audit_table`     | Glue table used for audit logs                           |
+| `audit_workgroup` | Athena workgroup used for Access Monitoring queries      |
+
+```json
```
Comment (r0mant) [inline docs/pages/access-controls/access-monitoring.mdx:64]: Let's move this under the collapsible "details" block.

File: docs/pages/access-controls/access-monitoring.mdx [hunk 2]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
+      ],
+      "Effect": "Allow",
+      "Resource": "arn:aws:kms:eu-central-1:1234567890:key/kms_id"
+    }
+  ],
+  "Version": "2012-10-17"
+}
+```
+
+You can used our [Terraform Example](https://github.com/gravitational/teleport/tree/f38bac4bcf6e462a1cab66a0b6074ded4201867d/examples/athena)
```
Comment (r0mant) [inline docs/pages/access-controls/access-monitoring.mdx:134]:
```suggestion
You can use our [Terraform Example](https://github.com/gravitational/teleport/tree/f38bac4bcf6e462a1cab66a0b6074ded4201867d/examples/athena)
```

File: docs/pages/access-controls/access-monitoring.mdx [hunk 3]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
+```bash
+$ terraform apply
+...
+ access_monitoring:
+   enabled: true
+   report_results: s3://long-term/report_results
+   role_arn: arn:aws:iam::123456789012:role/AccessMonitoringRole
+   workgroup: access_monitoring_workgroup
+...
+athena_url = "..."
```
Comment (r0mant) [inline docs/pages/access-controls/access-monitoring.mdx:146]: It is not clear how "access_monitoring" and "athena_url = ..." sections are related to "terraform apply" command. Is this the output from the apply command, or the example config users will need to set?

File: docs/pages/access-controls/access-monitoring.mdx [hunk 4]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
+   report_results: s3://audit-long-term/report_results
+   # (Optional) Athena workgroup used by access monitoring queries (if not set, the default primary workgroup will be used).
+   workgroup: access_monitoring_workgroup
 ```
 
-</TabItem>
-<TabItem scope={["cloud","team"]} label="Teleport Cloud">
 
-Teleport Access Monitoring is enabled by default for all Teleport Cloud users.
+Access Monitoring uses a different AWS role to execute Athena SQL queries.
```
Comment (r0mant) [inline docs/pages/access-controls/access-monitoring.mdx:48]: Nit: A role "different" from what? Might be helpful to clarify.

**Review by r0mant — APPROVED** [review]
lgtm with a few nits

File: docs/pages/access-controls/access-monitoring.mdx [hunk 5]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
 To enable Access Monitoring you need to update the Teleport Auth Service configuration:
 ```yaml
 auth_service:
-  access_monitoring:
-    enabled: true
-    # AWS role ARN that Teleport will assume to execute Athena SQL queries.
-    role_arn: arn:aws:iam::111111111111:role/access-monitoring-role
-    # S3 bucket where Access Monitoring reports will be stored.
-    report_results: s3://audit-log-example/report_results
+ access_monitoring:
```
Comment (nklaassen) [inline docs/pages/access-controls/access-monitoring.mdx:36]: I think this indent is only 1 space, probably should be 2

File: docs/pages/access-controls/access-monitoring.mdx [hunk 6]
```diff
@@ -23,28 +23,132 @@ Users are able to write their own custom access monitoring queries by querying t
...
+<TabItem scope={["cloud","team"]} label="Teleport Cloud">
+
+Teleport Access Monitoring is enabled by default for all Teleport Cloud users.
+
+</TabItem>
 <TabItem scope={["enterprise", "oss"]} label="Self-Hosted">
 
 To enable Access Monitoring you need to update the Teleport Auth Service configuration:
 ```yaml
 auth_service:
```
Comment (hugoShaka) [inline docs/pages/access-controls/access-monitoring.mdx:35]: Can you put this whole configuration block in an includable file and include it in the Teleport configuration reference https://goteleport.com/docs/reference/config/

**Review by hugoShaka — APPROVED** [review]
LGTM, once the remaining comments are addressed

**Pull request merged**